src/platform/nrfconnect/FactoryDataProvider.h

```diff
@@ -76,11 +76,21 @@ struct InternalFlashFactoryData
         // and make sure we do not overlap with settings partition
         constexpr size_t kFactoryDataBlockEnd =
             (FACTORY_DATA_ADDRESS + FACTORY_DATA_SIZE + CONFIG_FPROTECT_BLOCK_SIZE - 1) & (-CONFIG_FPROTECT_BLOCK_SIZE);
-        static_assert(kFactoryDataBlockEnd <= PM_SETTINGS_STORAGE_ADDRESS,
+
+        constexpr size_t kFactoryDataBlockBegin = FACTORY_DATA_ADDRESS & (-CONFIG_FPROTECT_BLOCK_SIZE);
```

Comment:
Because if I do this like this: constexpr size_t kFactoryDataBlockBegin = FactoryDataBlockBegin();                                         There is an error:                                                                                                                                                                                              
                                                                                                                                                                                              In member function 'constexpr size_t chip::DeviceLayer::InternalFlashFactoryData::FactoryDataBlockSize()':
../../../../../../../../modules/lib/matter/src/platform/nrfconnect/FactoryDataProvider.h:81:73: error: 'this' is not a constant expression
   81 |         constexpr size_t kFactoryDataBlockBegin = FactoryDataBlockBegin();
      |                                                                         ^
../../../../../../../../modules/lib/matter/src/platform/nrfconnect/FactoryDataProvider.h:89:23: error: non-constant condition for static assertion
   89 |         static_assert(kOverlapsCheck,

Reply:
Because if I do this like this: constexpr size_t kFactoryDataBlockBegin = FactoryDataBlockBegin();
there is an following error:
![image](https://github.com/user-attachments/assets/369db0c7-a4b8-493f-986f-9c6abbbe8b0d)